functionconvergence.go

```diff
@@ -0,0 +1,51 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package optimize
+
+import "math"
+
+// FunctionConvergence tests for the convergence of function values. See comment
+// in Settings.
+type FunctionConvergence struct {
+	Absolute   float64
+	Relative   float64
+	Iterations int
+
+	absBest float64
+	relBest float64
+	absIter int
+	relIter int
+}
+
+func (fc *FunctionConvergence) Init(f float64) {
+	fc.absBest = f
+	fc.relBest = f
+	fc.absIter = 0
+	fc.relIter = 0
+}
+
+func (fc *FunctionConvergence) FunctionConverged(f float64) Status {
+	if f >= fc.absBest-fc.Absolute {
```

Comment:
I would prefer this reorganized as in the relative test `fc.absBest - f <= fc.Absolute`.